### PR TITLE
Command structure refactor to match spec

### LINKED ISSUE #43: Refactor actions / modes
This task is to refactor the current action verbs to meet the [current Rover v2 spec](https://github.com/aztfmod/rovergo/wiki/Rover-v2-Draft-Proposal)

This involves moving the action from a command flag, to sub commands, one per action. This will result in many new sub-commands but should result in a logical structure. These sub-commands will all follow the same flow, the difference being the landingzone.Action they pass into the Execute()

The current semantics of rover v2 maps actions to terraform verbs almost 1:1, the exception being the action `deploy` which is a synonym for running `terraform plan` + `terraform apply`

I think switching back to the action named `apply` (which actually does a `terraform plan` + `terraform apply`) is best for now and will prevent confusion. However I'd welcome input and comments

### PR DESCRIPTION
**Pull request by benc-uk**
This PR is to refactor the current action verbs to meet the current Rover v2 spec

- 🎈 ADDS: New actions for all commands
- 🧶 CHANGES: the `landingzone` and `launchpad` commands dynamically build sub-commands from actions enum
- ❌ REMOVES: the shared RunFunc in landingzone/command.go replaced by new dynamic process

Fixes #43

### CODE REVIEW

File: pkg/landingzone/landingzone.go [hunk 1]
```diff
@@ -136,16 +139,14 @@ func (o *Options) Execute(action Action) {
...
+			console.Success("Any apply step will be skipped")
 		}
 	}
 
 	//
-	// Deploy action
+	// Terraform apply step, won't run if plan found no changes
 	//
-	if action == ActionDeploy {
+	if (action == ActionApply || action == ActionRun) && planChanges {
```
Comment (sebastus) [inline pkg/landingzone/landingzone.go:149]: Looks like we lost the validate phase?

**Review by sebastus — CHANGES_REQUESTED** [review]
One thing...

File: pkg/landingzone/landingzone.go (hunk 1 above)
Comment (benc-uk) [reply pkg/landingzone/landingzone.go]: Good catch!!! Fixed

**Review by sebastus — APPROVED** [review]
lgtm

**Pull request merged**